pkg/pillar/types/certinfotypes.go

```diff
@@ -6,22 +6,43 @@ package types
 import (
 	"encoding/hex"
 
-	zcert "github.com/lf-edge/eve/api/go/certs"
-	zcommon "github.com/lf-edge/eve/api/go/evecommon"
+	"github.com/google/go-tpm/tpmutil"
+	"github.com/lf-edge/eve/api/go/evecommon"
 )
 
-// ControllerCert : controller certicate
-// config received from controller
-type ControllerCert struct {
-	HashAlgo zcommon.HashAlgorithm
-	Type     zcert.ZCertType
-	Cert     []byte
-	CertHash []byte
-	// ErrorAndTime provides SetErrorNow() and ClearError()
+type ZCertOrigin uint8
+
+const (
+	CERT_ORIGIN_NONE ZCertOrigin = iota + 0
+	CERT_ORIGIN_CONTROLLER
+	CERT_ORIGIN_EVE_NODE_SOFTWARE
+	CERT_ORIGIN_EVE_NODE_TPM
+	CERT_ORIGIN_EVE_NODE_HSM
+)
+
+// ZCertConfig : basic certificate config structure
+type ZCertConfig struct {
+	HashAlgo  evecommon.HashAlgorithm
+	Type      evecommon.ZCertType
+	Origin    ZCertOrigin
+	TpmHandle tpmutil.Handle
+	Hash      []byte
+	Cert      []byte
+	PvtKey    []byte
```

Comment:
A private key makes no sense as I commented in the other PR. When you need to do an operation the operation is performed (using the TPM with its private key, or locally loading a private key from file if there is no TPM). But a private key never belongs in any go struct.